File: dashboard-backend/src/main/java/com/quanluu/dashboard/service/TheoremService.java
package com.quanluu.dashboard.service;

import com.quanluu.dashboard.model.dto.TheoremDTO;

import java.util.List;

public interface TheoremService {

    List<TheoremDTO> getAllTheorems();

    TheoremDTO getTheoremById(Long id);

    List<TheoremDTO> searchTheoremsByTitleContaining(String keywords, Integer limit);

    TheoremDTO getMostRecentTheorem();

    TheoremDTO createTheorem(TheoremDTO theoremDTO);

    TheoremDTO updateTheorem(Long id, TheoremDTO theoremDTO);

    void deleteTheorem(Long id);
}

File: dashboard-frontend/src/services/theoremService.js
import axios from 'axios'

const API_URL = 'http://localhost:8080/api/theorems'

export default {
  getAllTheorems() {
    return axios.get(API_URL)
  },

  getTheoremById(id) {
    return axios.get(`${API_URL}/${id}`)
  },

  searchTheorems(searchText) {
    const keywords = searchText.trim().split(/\s+/).join(',')
    return axios.get(`${API_URL}/search`, {
      params: { keywords, limit: 5 }
    })
  },

  getMostRecentTheorem() {
    return axios.get(`${API_URL}/recent`)
  },

  createTheorem(theorem) {
    return axios.post(API_URL, theorem)
  },

  updateTheorem(id, theorem) {
    return axios.put(`${API_URL}/${id}`, theorem)
  },

  deleteTheorem(id) {
    return axios.delete(`${API_URL}/${id}`)
  }
}
